### DISCUSSION TITLE
Use Neovim for true Vim Editor backend

### DISCUSSION THREAD

**Discussion by bugeats** [standardnotes/forum#2536] [Suggestions]
[Neovim](https://neovim.io/) takes vim and splits it into a client/server model (among other things). It's a perfect fit for adding a true vim experience to Standard Notes.

**Comment by atmoio** [comment]
The thing is though, if it requires handling user inputted content on a server, this would break the client-privacy model of current extensions (what they type does not leave their computer). But Neovim looks like a native desktop client anyway?

**Comment by bugeats** [comment]
Oh there’s no remote server involved. Neovim is just splitting the core engine from the presentation rendering. Buffers would remain in memory on the local machine.

Take a deeper look at the project. There’s been several UIs written for Neovim, and many of them (too many IMO) are using Electron.

**Comment by filyp** [comment]
Bump.
I miss all the vim goodies too. Adding them would be an enormous improvement to usability for me.

**Comment by krillin666** [comment]
Bump.

VIM keybindings would be a crucial feature for SN.
Also keep up the amazing work !
